### DISCUSSION TITLE
Unintended change of date

### DISCUSSION THREAD

**Discussion by burningT** [TechbeeAT/jtxBoard#1334] [Q&A]
**Describe the bug**
When a task contains two date entries with no time and one of them is modified, the other one is added with the time 0:00.

**To Reproduce**
1. Create and dave a task with start date 2024-04-08 and due date 2024-04-09. Ensure no time is set.
2. Change and save the due date to 2024-04-10.
4. See start time being 2024-04-08 0:00

**Expected behavior**
Only the inzended date change occurs.

**Device and version**
 - SHIFT6mq
 - Android version: 10
 - jtx Board version: 2.07.03-beta03.gplay
 - Downloaded from: Google Play

**Comment by patrickunterwegs** [comment]
Hi @burningT , this is not a bug, regrettably it's a feature ;-)
I know that it might sound stupid, but it's the iCalendar standard that enforces, that both dates, start and due, must have the same format, either date only or date with time. That's why I add 00:00 to the other date when you set it in one (or remove it in both if you remove it for one). 
Even if I would allow it, servers might reject such entries that don't follow this rule. That's why it's just better to stick to the rule, even though it might not be intuitive for users...

**Reply by burningT** [reply]
I might be missung something here. You mention that both dates need to have the same format. So if you would add a time to one date, you have to add a time to the other date as well. How is that related to my case, where both dates have no time at all and I just change one of them without adding a time?

**Reply by patrickunterwegs** [reply]
Hm, sorry for the late reply!
And sorry, I might have misunderstood your initial description. My statement is basically valid, but what you describe shouldn't happen as it would even cause the conflict (instead of preventing it).

**Reply by patrickunterwegs** [reply]
Hi @burningT , I just tried to reproduce this issue, but for me it seems to work. Can you always reproduce the problem?

**Reply by burningT** [reply]
Hi,
I could indeed in the past. However, I tried it now again and it seems changed. I can still reproduce if I change step 2:  Disable the date instead of changing it.

**Reply by patrickunterwegs** [reply]
Thanks @burningT ! 
Indeed, that's a problem , I just reproduced and it. I'll make a fix soon!
